libraries/I2S/src/I2S.cpp

```diff
@@ -199,13 +250,18 @@ void I2S::end() {
 }
 
 int I2S::available() {
-    if (!_running) {
+    if (!_running) { 
         return 0;
     } else {
         auto avail = _arb->available();
         avail *= 4; // 4 samples per 32-bits
-        if (_bps < 24 && !_isOutput) {
-            avail += _isHolding / 8;
+        if (_bps < 24) {
+            if (_isOutput) {
+                // 16- and 8-bit can have holding bytes available
+                avail += (32 - _isHolding) / 8;
+            } else {
+                avail += _isHolding / 8;
+            }
```

Comment:
I think you made a merge mistake here. You're undoing my #1548

Reply:
Reinserted code into pull request as per #1548